[aoti_debug_printer][BE] explicitly dumping float32, bfloat16, float16 data type

YUNQIUGUO opened the pull request:
Summary:
per request, explicitly dumping the float dtypes for aten tensors in debug printing summary info.

can be useful in identifying issues such as "wrong AOTI Lowering precisions"

Test Plan:
```
 AOT_INDUCTOR_DEBUG_INTERMEDIATE_VALUE_PRINTER=2 TORCH_LOGS="+inductor, output_code" buck2 run -c fbcode.enable_gpu_sections=true -c fbcode.nvcc_arch=h100 @//mode/opt fbcode//caffe2/test/inductor:test_aot_inductor -- -r test_addmm
```

Differential Revision: D69547344




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

jingsh reviewed (approved):
lgtm

ColinPeppler reviewed (changes requested):
to add prints for integer types too

YUNQIUGUO committed:
[aoti_debug_printer][BE] explicitly dumping float32, bfloat16, float16 data type (#147020)

Summary:

Per request, explicitly dumping the float dtypes for aten tensors in debug printing summary info.

can be useful in identifying issues associated with "wrong AOTI lowering input precisions"

before this diff:
[  before_launch - triton_tem_fused_addmm_1 - buf1  ]:
Number of elements: 6144
**Dtype: Float**
Mean value: 0.0195527
Min value: 0
Max value: 4.55529
Device: cuda:0
Size: [1024, 6]
Stride: [6, 1]
Layout: Strided
Is contiguous: 1
Requires grad: 0


after this diff:
[  before_launch - triton_tem_fused_addmm_1 - buf1  ]:
Number of elements: 6144
**Dtype: float32**
Mean value: 0.0195527
Min value: 0
Max value: 4.55529
Device: cuda:0
Size: [1024, 6]
Stride: [6, 1]
Layout: Strided
Is contiguous: 1
Requires grad: 0

Test Plan:
```
 AOT_INDUCTOR_DEBUG_INTERMEDIATE_VALUE_PRINTER=2 TORCH_LOGS="+inductor, output_code" buck2 run -c fbcode.enable_gpu_sections=true -c fbcode.nvcc_arch=h100 @//mode/opt fbcode//caffe2/test/inductor:test_aot_inductor -- -r test_addmm
```

Reviewed By: hl475, jingsh, ColinPeppler

Differential Revision: D69547344
